cores/esp32/Esp.cpp

```diff
@@ -199,10 +204,10 @@ bool EspClass::flashEraseSector(uint32_t sector)
 
 bool EspClass::flashWrite(uint32_t offset, uint32_t *data, size_t size)
 {
-    return spi_flash_write(offset, (uint32_t*) data, size) == ESP_OK;
+    return spi_flash_write_encrypted(offset, (uint32_t*) data, size) == ESP_OK;
```

Comment:
This will unconditionally encrypt and write to flash, with the restriction that writes must be 16-byte aligned. This may not be always desired, for example a file system like SPIFFS will try to write smaller chunks of data and fail due to this restriction. Also NAND way of writing (i.e. flipping 1s to 0s) will not work with spi_flash_write_encrypted.

I suggest adding APIs which specifically do encrypted writes (or even better, adding a library for this purpose).